plugins/package-managers/maven/src/main/kotlin/utils/LocalRepositoryHelper.kt

```diff
@@ -43,6 +48,44 @@ class LocalRepositoryHelper(
 
         /** The name of the folder that stores OSGi bundles. */
         private const val BUNDLE_ROOT = "osgi/bundle"
+
+        /** The name of the XML file with properties for P2 artifacts contained in the local repository. */
+        private const val ARTIFACTS_FILE = "p2artifacts.xml"
+
+        /**
+         * Parse the given [file] with information about artifacts and extract the properties for the given [artifact].
+         */
+        private fun parseArtifactsFile(file: File, artifact: Artifact): Map<String, String> {
+            val properties = mutableMapOf<String, MutableMap<String, String>>()
+
+            val handler = object : DefaultHandler() {
+                private var currentArtifactId = ""
+
+                override fun startElement(uri: String?, localName: String, qName: String, attributes: Attributes) {
+                    when (qName) {
+                        "artifact" -> {
+                            currentArtifactId = "${artifact.groupId}:${attributes.getValue("id")}:" +
+                                "${attributes.getValue("version")}"
+                        }
+
+                        "property" -> {
+                            val key = attributes.getValue("name")
+                            val value = attributes.getValue("value")
+
+                            properties.getOrPut(currentArtifactId) { mutableMapOf() }[key] = value
```

Comment:
Nit: Consider extracting the `properties.getOrPut(currentArtifactId) { mutableMapOf() }` part to a variable for readability.